File: src/main/java/org/example/expert/domain/user/controller/UserController.java
package org.example.expert.domain.user.controller;


import lombok.RequiredArgsConstructor;
import org.example.expert.domain.common.dto.AuthUser;
import org.example.expert.domain.user.dto.request.UserChangePasswordRequest;
import org.example.expert.domain.user.dto.response.UserResponse;
import org.example.expert.domain.user.service.UserService;
import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;

    @GetMapping("/users/{userId}")
    public ResponseEntity<UserResponse> getUser(@PathVariable long userId) {
        return ResponseEntity.ok(userService.getUser(userId));
    }

    @PutMapping("/users")
    public void changePassword(@AuthenticationPrincipal AuthUser authUser, @RequestBody UserChangePasswordRequest userChangePasswordRequest) {
        userService.changePassword(authUser.getId(), userChangePasswordRequest);
    }

    @GetMapping("/users/all")
    public ResponseEntity<Page<UserResponse>> getAllUser(
        @RequestParam(defaultValue = "1") int page,
        @RequestParam(defaultValue = "10") int size,
        @RequestParam(required = false) String username
    ) {
        return ResponseEntity.ok(userService.getAllUser(page, size, username));
    }

//    @GetMapping("/users/all")
//    public ResponseEntity<List<UserResponse>> getAllUser(
//        @RequestParam(required = false) Long lastUserId,
//        @RequestParam(defaultValue = "10") int size,
//        @RequestParam(required = false) String username
//    ) {
//        return ResponseEntity.ok(userService.getAllUser(lastUserId, size, username));
//    }


}

File: src/main/java/org/example/expert/domain/user/controller/UserController.kt
package org.example.expert.domain.user.controller

import lombok.RequiredArgsConstructor
import org.example.expert.domain.common.dto.AuthUser
import org.example.expert.domain.user.dto.request.UserChangePasswordRequest
import org.example.expert.domain.user.dto.response.UserResponse
import org.example.expert.domain.user.service.UserService
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*
import org.springframework.data.domain.Page

@RestController
@RequiredArgsConstructor
class UserController (
    private val userService: UserService
) {

    @GetMapping("/users/{userId}")
    fun getUser(@PathVariable userId:Long):ResponseEntity<UserResponse> {
        return ResponseEntity.ok(userService.getUser(userId))
    }

    @PutMapping("/users")
    fun changePassword(@AuthenticationPrincipal authUser: AuthUser,
                       @RequestBody userChangePasswordRequest: UserChangePasswordRequest)
    {
        userService.changePassword(authUser.id,userChangePasswordRequest)
    }

    @GetMapping("/users/all")
    fun getAllUser(@RequestParam(defaultValue = "1") page:Int,
                   @RequestParam(defaultValue = "10") size:Int,
                   @RequestParam(required = false) username:String
                   ): ResponseEntity<Page<UserResponse?>?> {
        return ResponseEntity.ok(userService.getAllUser(page,size, username))
    }

//    @GetMapping("/users/all")
//    fun getAllUser(@RequestParam(required = false) lastUserId:Long,
//                   @RequestParam(defaultValue = "10") size:Int,
//                   @RequestParam(required = false) username: String
//    ): ResponseEntity<List<UserResponse?>?> {
//        return ResponseEntity.ok(userService.getAllUser(lastUserId,size,username))
//    }

}